cmd/remote_val_harness/internal/test_harness.go

```diff
@@ -194,6 +201,56 @@ func (th *TestHarness) TestSignProposal() error {
 	return nil
 }
 
+func (th *TestHarness) TestSignVote() error {
+	th.logger.Info("TEST: Signing of votes")
+	for _, voteType := range voteTypes {
+		th.logger.Info("Testing vote type", "type", voteType)
+		// sha256 hash of "hash"
+		hash, _ := hex.DecodeString("D04B98F48E8F8BCC15C6AE5AC050801CD6DCFD428FB5F9E65C4E16E7807340FA")
+		vote := &types.Vote{
+			Type:   voteType,
+			Height: 12345,
+			Round:  23456,
+			BlockID: types.BlockID{
+				Hash: hash,
+				PartsHeader: types.PartSetHeader{
+					Hash:  hash,
+					Total: 1000000,
+				},
+			},
+			ValidatorIndex:   0,
+			ValidatorAddress: hash[:20],
+			Timestamp:        time.Now(),
+		}
+		// work out the canonicalized serialized byte form of the message
+		// without its signature
+		voteBytes, err := cdc.MarshalBinaryLengthPrefixed(vote)
```

Comment:
Unrelated to the problem you are seeing on the kms side, this should be vote.SignBytes(th.chainID). Otherwise the verficiation below will fail.